java/impl/src/main/java/net/sf/mmm/code/impl/java/JavaContext.java

```diff
@@ -195,15 +197,19 @@ public BaseContext getContext() {
     @Override
     public BaseType getType(String qualifiedName) {
 
+      Class<?> clazz = null;
       try {
-        Class<?> clazz = this.classloader.loadClass(qualifiedName);
+        clazz = classloader.loadClass(qualifiedName);
         if (clazz.isArray()) {
           throw new IllegalArgumentException(qualifiedName);
         }
         return (BaseType) getContext().getType(clazz);
       } catch (ClassNotFoundException e) {
         LOG.debug("Class {} not found.", qualifiedName, e);
         return null;
+      } catch (NoClassDefFoundError e) {
+        LOG.debug("Failed to find a dependency.", e);
```

Comment:
> getType("none.existing.Type"). So LOG.debug("Could not find class {}.", qualifiedName); would make more sense.

I agree, I have changed it.

> but must not be a failing dependency

Actually, it really is a failing dependency problem. I got this problem because I removed one of my jars on the `.m2` folder. The issue is that Eclipse uses its target folder to compile, but maybe the jar is missing on the local repository. The JavaDoc of `NoClassDefFoundError` is the following:

> Thrown if the Java Virtual Machine or a ClassLoader instance tries to load in the definition of a class (as part of a normal method call or as part of creating a new instance using the new expression)and no definition of the class could be found. The searched-for class definition existed when the currently executing class was compiled, but the definition can no longer be found

Therefore, I would still leave the message.